src/lib/promiseHelper.js

```diff
@@ -18,3 +18,5 @@ export const resolvePromiseProperties = (obj) => {
     // resolved values
     Object.assign({}, obj, zipObject(promiseKeys, resolvedPromises)));
 };
+
+export const then = func => p => ((p && p.then) ? p.then(func) : func(p));
```

Comment:
This is a very simple function that lets you do something like the following:

```js
const fn = flow([
  then(x => x + 2), // this will work whether or not x is a Promise
])
fn(2) // -> 4
fn(Promise.resolve(2)) // -> Promise resolving to 4
```
One notable question: should this return synchronously if it doesn't receive a Promise  (as it does currently), or should it be rewritten as:

```js
export const then = func => p => Promise.resolve(p).then(func)
```

The advantage of this would be that `then` would always *return* a Promise, regardless of what was passed in, which seems more in line with the name and analogy to the `.then` method it's wrapping.